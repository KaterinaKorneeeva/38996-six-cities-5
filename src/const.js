export const CITY_LIST = [`Paris`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const SortingType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};
