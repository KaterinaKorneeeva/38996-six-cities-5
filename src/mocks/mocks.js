export const offerList =
   [
     {
       city: {
         name: `Cologne`,
         location: {
           latitude: `50.938361`,
           longitude: `6.959974`,
           zoom: `13`
         }
       },
       previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
       images: [
         `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
         `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
         `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
       ],
       title: `The house among olive `,
       isFavorite: true,
       isPremium: false,
       rating: 3.8,
       type: `house`,
       bedrooms: 3,
       maxAdults: 4,
       price: 493,
       goods: [
         `Laptop friendly workspace`,
         `Washer`,
         `Towels`,
       ],
       host: {
         id: 25,
         name: `Angelina`,
         isPro: true,
         avatarUrl: `img/avatar-angelina.jpg`,
         avatar: `img/avatar-angelina.jpg`
       },
       description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
       location: {
         latitude: 50.957361,
         longitude: 6.9509739999999995,
         zoom: 16
       },
       id: 1
     }
   ];
export const comments =
  [
    {
      id: 1,
      user: {
        id: 11,
        isPro: false,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`
      },
      rating: 5,
      comment: `This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.`,
      date: `2020-10-10T13:38:44.753Z`
    },
    {
      id: 2,
      user: {
        id: 11,
        isPro: false,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`
      },
      rating: 5,
      comment: `This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.`,
      date: `2020-10-10T13:38:44.753Z`
    },
  ];
