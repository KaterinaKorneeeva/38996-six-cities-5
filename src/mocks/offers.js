import {getRandomInteger} from "../utils.js";

const PIC_URL = `https://picsum.photos/200/300?random=`;
const AVATAR_URL = `https://api.adorable.io/avatars`;

export default [
  {
    id: 0,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(1),
    title: `Wood and stone place`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    price: `60`,
    type: `room`,
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    rooms: `4`,
    guests: `3`,
    features: [
      `wifi`,
      `heating`,
      `kitchen`,
      `cableTV`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    host: {
      name: `Simon`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
          },
        data: `2020-10-10`,
      },
      {
        text: `We booked at very short notice and Simon was quick to respond which was very helpful a long with details on how to check in. Accommodation is in a very good spot, within walking distance to tourist attractions like London eye etc. Apartment looked bigger in photos but was very cosy and had what we needed, would have loved if there was a tv even for some back ground noise. Over all good experience :)`,
        author:
          {
            name: `Annie`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
          },
        data: `2020-10-09`,
      }
    ]
  },
  {
    id: 1,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(0),
    title: `Beautiful & luxurious apartment at great location`,
    description: `The hostel serves a daily breakfast buffet and a set menu for dinner. There is also a bar in the hostel. There are many supermarkets in the surrounding streets, trendy bars, restaurants and shops in various squares in the Gr√†cia area, 15 minutes' walk away. The nearest Metro Station is Verdaguer, 500 meters from the hostel.`,
    price: `120`,
    type: `Apartment`,
    city: `Amsterdam`,
    coords: [52.369553943508, 4.85309666406198],
    rooms: `4`,
    guests: `4`,
    features: [`wifi`, `heating`, `kitchen`, `cableTV`],
    host: {
      name: `Angelina`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,

            rating: `4`,
          },
        data: `2020-10-10`,

      }
    ]
  },
  {
    id: 2,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(1),
    title: `Nice, cozy, warm big bed apartment`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    price: `180`,
    type: `house`,
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.929309666406198],
    rooms: `4`,
    guests: `4`,
    features: [
      `wifi`,
      `heating`,
      `kitchen`,
      `cableTV`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    host: {
      name: `Angelina`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
            rating: `4`,
          },
        data: `2020-10-10`,

      }
    ]
  },
  {
    id: 3,
    rating: `5`,
    pictures: [
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
      `${PIC_URL}${Math.random()}`,
    ],
    isPremium: Boolean(0),
    title: `Nice, cozy, warm big bed apartment`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    price: `180`,
    type: `hotel`,
    city: `Brussels`,
    coords: [52.3809553943508, 4.939309666406198],
    rooms: `4`,
    guests: `4`,
    features: [`wifi`, `heating`, `kitchen`, `cableTV`],
    host: {
      name: `Angelina`,
      status: `Pro`,
      photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
    },
    reviews: [
      {
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        author:
          {
            name: `Max`,
            photo: `${AVATAR_URL}/${getRandomInteger(1, 100)}`,
            rating: `4`,
          },
        data: `April 2019`,

      }
    ]
  }
];
