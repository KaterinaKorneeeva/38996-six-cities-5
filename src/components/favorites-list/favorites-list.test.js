import React from "react";
import renderer from "react-test-renderer";
import FavoritesList from "./favorites-list";
import {offerList} from "./../../mocks/mocks";

import {MemoryRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import {AuthorizationStatus} from "../../const";
import {Provider} from "react-redux";

const mockStore = configureStore()({
  USER: {
    authorizationStatus: AuthorizationStatus.AUTH
  }
});

it(`Should FavoritesList render correctly`, () => {
  const noop = () => {};
  const city = `Amsterdam`;
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <MemoryRouter>
            <FavoritesList
              city = {city}
              onOfferCardHover = {noop}
              handleFavoriteClick = {noop}
              favoriteOffersInCity = {offerList}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
