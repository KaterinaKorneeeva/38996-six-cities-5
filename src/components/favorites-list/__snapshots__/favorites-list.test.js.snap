// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesList render correctly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="near-places__card place-card"
      onMouseOver={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              493
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button--active button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "76%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            The house among olive 
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          house
        </p>
      </div>
    </article>
  </div>
</li>
`;
